### PR TITLE
Update vision docstring bool masked pos

### LINKED ISSUE #21484: Add/fix documentation around VideoMAEForPretraining's `bool_masked_pos` argument
### System Info

N/A

### Who can help?

@NielsRogge @amyeroberts 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

The `VideoMAEForPretraining` class is missing a docstring for an important argument to the `forward` function, `bool_masked_pos`. Can see this in the docs for the `main` branch [here](https://huggingface.co/docs/transformers/main/en/model_doc/videomae#transformers.VideoMAEForPreTraining.forward)...it's listed in the function signature but not in the documented arguments.

There is an [example code snippet](https://huggingface.co/docs/transformers/main/en/model_doc/videomae#transformers.VideoMAEForPreTraining.forward.example) that creates this input tensor that you can see in the docs, but I'm not sure it's correct when applied to video inputs with `batch_size > 1` (if not done carefully). When I calculated it example-by-example, I was getting errors, which is what led me to this issue.

I dug a little deeper and noticed that the test suite for VideoMAE has [different logic](https://github.com/huggingface/transformers/blob/5b49376202863d3798d2ff8a8ba61590542a1141/tests/models/videomae/test_modeling_videomae.py#L145-L149) for creating `bool_masked_pos`. When I updated my training code to use that logic, my problems went away. I assume this is related to the note in the tests that mentions each video needing the same number of masked patches.

### Expected behavior

To resolve this issue, I think we should:

1. Add a docstring for `bool_masked_pos` that explains what it is for users
2. Update the example snippet to use the correct logic from the test suite.

Even better I guess would be to handle this for users in the `ImageProcessor`, but I'll leave that for a separate issue.

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Add the missing `bool_masked_pos` information in the docstring for vision models.

Fixes #21484 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/models/beit/modeling_beit.py [hunk 1]
```diff
@@ -659,6 +659,10 @@ def forward(
         output_hidden_states: Optional[bool] = None,
         return_dict: Optional[bool] = None,
     ) -> Union[tuple, BeitModelOutputWithPooling]:
+        r"""
+        bool_masked_pos (`torch.BoolTensor` of shape `(batch_size, num_patches)`, *optional*):
+                    Boolean masked positions. Indicates which patches are masked (1) and which aren't (0).
```
Comment (nateraw) [inline src/transformers/models/beit/modeling_beit.py:664]: Is this overindented or no? (not familiar with autodocstring magic logic)

**Review by nateraw — APPROVED** [review]
LGTM thanks so much

File: src/transformers/models/beit/modeling_beit.py (hunk 1 above)
Comment (amyeroberts) [reply src/transformers/models/beit/modeling_beit.py]: It is! I'd just lazily copy-pasted, so I've fixed here and in the original place

**Review by sgugger — APPROVED** [review]
Thanks for adding those!
